internal/file/group.go

```diff
@@ -84,3 +77,50 @@ func (fileGroup *Group) getFilePathDependantCases() []string {
 		".pip.debricked.lock",
 	}
 }
+
+func (fileGroup *Group) matchLockFile(lockFile, dir string) bool {
+	if !fileGroup.HasFile() {
+
+		return false
+	}
+	groupDir, manifestFile := filepath.Split(fileGroup.ManifestFile)
+	if groupDir != dir {
+
+		return false
+	}
+
+	return matchFile(manifestFile, lockFile)
+}
+
+func (fileGroup *Group) matchManifestFile(manifestFile, dir string) bool {
+	if fileGroup.HasFile() {
+
+		return false
+	}
+	groupDir, lockFile := filepath.Split(fileGroup.LockFiles[0])
+	if groupDir != dir {
+
+		return false
+	}
+
+	return matchFile(manifestFile, lockFile)
+}
+
+func matchFile(manifestFile, lockFile string) bool {
+	var isPIP, isNPM bool
+	lockFile, isPIP = strings.CutSuffix(lockFile, ".pip.debricked.lock")
+	if isPIP {
+		lockFile, _ = strings.CutPrefix(lockFile, ".")
+
+		return lockFile == manifestFile
+	}
+
+	lockFile, isNPM = strings.CutSuffix(lockFile, "-lock.json")
+	if isNPM {
+		manifestFile, _ = strings.CutSuffix(manifestFile, ".json")
+
+		return lockFile == manifestFile
+	}
```

Comment:
I want to read up on these cases, could you give me a hint as to where I can do that?

Reply:
As I said in another comment, I seem to be mistaken. The NPM code is removed now since I can't find the source that described this scenario.